plugins/obs-qsv11/common_directx11.cpp

```diff
@@ -224,7 +226,7 @@ mfxStatus _simple_alloc(mfxFrameAllocRequest *request,
 		desc.Format = format;
 		desc.SampleDesc.Count = 1;
 		desc.Usage = D3D11_USAGE_DEFAULT;
-		desc.BindFlags = D3D11_BIND_DECODER;
+		desc.BindFlags = D3D11_BIND_VIDEO_ENCODER;
```

Comment:
Have we had this wrong the whole time? Was this caught by the D3D debug layer?

Reply:
Fwiw, Intel still uses D3D11_BIND_DECODER in their sample. Do they have a bug?
https://github.com/Intel-Media-SDK/MediaSDK/blob/34ec1df8448c1f26dfd25a5fa54f4c0486d479b7/tutorials/common/common_directx11.cpp#L225

Reply:
Yes, I think so. A Bug.
https://docs.microsoft.com/en-us/windows/win32/api/d3d11/ne-d3d11-d3d11_bind_flag

D3D11_BIND_DECODER: The common way to create resources for a **decoder output** is by calling the ID3D11Device::CreateTexture2D method to create an array of 2D textures.

D3D11_BIND_VIDEO_ENCODER: The common way to create **resources for a video encoder** is by calling the ID3D11Device::CreateTexture2D method to create an array of 2D textures.

Reply:
I know from the name, it looks like a bug, but maybe Intel has a special reason. I'll get separate clarification from them.

Reply:
Seems like D3D11_BIND_VIDEO_ENCODER is being omitted on purpose. This is what Intel's code looks like, which I've copied into the PR:
```
#if (MFX_VERSION >= 1025)
        if ((request->Type&MFX_MEMTYPE_VIDEO_MEMORY_ENCODER_TARGET) && (request->Type & MFX_MEMTYPE_INTERNAL_FRAME))
        {
            desc.BindFlags = D3D11_BIND_DECODER | D3D11_BIND_VIDEO_ENCODER;
        }
        else
#endif
            desc.BindFlags = D3D11_BIND_DECODER;
```